py/builtinimport.c

```diff
@@ -521,6 +539,8 @@ STATIC mp_obj_t process_import_at_level(qstr full_mod_name, qstr level_mod_name,
         // a __path__ attribute, and not attempt to stat it.
     }
 
+    nlr_pop_jump_callback(false);
```

Comment:
Should this be moved later, in case the `mp_store_attr` below fails (due to out-of-memory)?  Not sure...

Reply:
Yeah, it probably makes sense for the module to be removed if anything goes wrong during the rest of the `process_import_at_level` function.